### PR TITLE
feat!: use data factory testing framework expressions evaluator

### PR DESCRIPTION
**Pull request by LeonardHd**
- [x] update documentation (requirements, features)
- [x] discuss naming of components

### CODE REVIEW

File: src/data_factory_testing_framework/_expression_runtime/data_factory_expression/data_factory_to_dotnet_evaluator_transformer.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from typing import Union
+
+from lark import Token, Transformer, Tree, v_args
+
+
+@v_args(tree=True)
+class DataFactoryToDotnetEvaluatorExpressionTransformer(Transformer):
```
Comment (arjendev) [inline src/data_factory_testing_framework/_expression_runtime/data_factory_expression/data_factory_to_dotnet_evaluator_transformer.py:7]: Should this be the DataFactoryToLogicAppsExpressionTransformer?

File: src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py [hunk 2]
```diff
@@ -86,15 +79,15 @@ def __init__(self) -> None:
             EXPRESSION_LOGICAL_BOOL: "or" | "and"
             EXPRESSION_NULL: NULL
             EXPRESSION_PARAMETER_NAME: /[a-zA-Z0-9_]+/
-            EXPRESSION_PIPELINE_PROPERTY.2: "parameters" | "globalParameters"
+            EXPRESSION_PIPELINE_PROPERTY: /[a-zA-Z0-9_]+/
```
Comment (arjendev) [inline src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py:82]: What's the reason of dropping the priority?

File: src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py [hunk 3]
```diff
@@ -45,38 +39,37 @@ def __init__(self) -> None:
...
-            expression_pipeline_reference: "pipeline" "()" "." EXPRESSION_PIPELINE_PROPERTY "." EXPRESSION_PARAMETER_NAME
             expression_variable_reference: "variables" "(" EXPRESSION_VARIABLE_NAME ")"
-            expression_activity_reference: "activity" "(" EXPRESSION_ACTIVITY_NAME ")"
-            expression_dataset_reference: "dataset" "()" "." EXPRESSION_PARAMETER_NAME
-            expression_linked_service_reference: "linkedService" "()" "." EXPRESSION_PARAMETER_NAME
+            expression_datafactory_parameters_reference: EXPRESSION_DATAFACTORY_REFERENCE "()"
+            expression_datafactory_activity_reference: "activity" "(" EXPRESSION_ACTIVITY_NAME ")"
             expression_item_reference: "item" "()"
-            expression_system_variable_reference: "pipeline" "()" "." EXPRESSION_SYSTEM_VARIABLE_NAME
+            expression_parameter_reference: "parameter" "(" EXPRESSION_PARAMETER_NAME ")"
```
Comment (arjendev) [inline src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py:57]: Is this expression still used? It's not data factory expression language afaik and if it's logic apps expression language I don't see any transformations using this. Curious!

File: src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py (hunk 2 above)
Comment (arjendev) [reply src/data_factory_testing_framework/_expression_runtime/data_factory_expression/expression_transformer.py:82]: I see now, you merge any pipeline().<parameter> into a single rule and leverage the logic apps runtime to approach pipeline().parameters.<name> and pipeline().globalParamters.<name> just like any other pipeline().<name>

File: src/data_factory_testing_framework/_expression_runtime/data_factory_expression/data_factory_to_dotnet_evaluator_transformer.py (hunk 1 above)
Comment (LeonardHd) [reply src/data_factory_testing_framework/_expression_runtime/data_factory_expression/data_factory_to_dotnet_evaluator_transformer.py]: Renamed to be aligned with our published nuget `Microsoft.Azure.DataFactoryTestingFramework.Expressions`

**Comment by arjendev** [comment]
The limitations in README.md can be revised

File: src/data_factory_testing_framework/_pythonnet/Evaluator.csproj [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (LeonardHd) [reply src/data_factory_testing_framework/_pythonnet/Evaluator.csproj:3]: I would recommend installing dotnet 8.0 (it's at least current) but we can still support 5.0.
I will adjust the documentation accordingly.

File: examples/data_factory/xml/test_xml.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import base64
+
+import pytest
```
[the remark this answers is not in the document]
Comment (LeonardHd) [reply examples/data_factory/xml/test_xml.py:3]: Moved the files to functional tests. I agree with your argument regarding examples vs. test.

File: pyproject.toml [hunk 6]
```diff
@@ -1,24 +1,38 @@
-[tool.poetry]
+[project]
```
[the remark this answers is not in the document]
Comment (LeonardHd) [reply pyproject.toml:1]: As agreed,
- author-email (as via setuptools generated should be good enough).
- will use the setuptools / pypi project urls as from the example <https://pypi.org/project/links-demo/>
- added classifiers as of [Classifiers](https://pypi.org/classifiers/)

**Review by arjendev — APPROVED** [review]
Let's merge on Monday! Enjoy the weekend!

**Pull request merged**